### ISSUE TITLE
Add -e cli flag

### ISSUE THREAD

**Issue by kyranet** [denoland/deno#1038]
From [Node.js docs](https://nodejs.org/dist/latest-v10.x/docs/api/cli.html#cli_e_eval_script), it may be done after #779, it helps with quick debugging such as inspecting the output of a one-liner.

**Comment by rsp** [comment]
Just for a different perspective than only doing what Node does, here is how `-e` plus some complementary switches work in Perl:

 `-e` evals the string like if it was a normal program (no implicit print or anything else)
 `-n` runs the code* for all lines of standard input**
 `-p` runs the code* for all lines of standard input** and prints the result
 `-l` strips newlines from the input and adds newlines for print

*) the code is the value of `-e` if present or a normal program in a file otherwise (i.e. `-p` and other switches don't imply `-e` and in fact you can use `-p` or `-n` in a shebang lines of normal scripts which is super useful)

**) the input is actually stdin if no arguments are provided, or it opens file(s) if there are arguments (just like `cat`, `sort`, `uniq` and other Unix filters traditionally work)

In Perl the input and output is in the `$_` variable (which is default for regex substitutions, print etc.), maybe in Deno some other way of providing the lines would make more sense - e.g. few years ago I created [filt](https://www.npmjs.com/package/filt) on npm to use Node for simple filters instead of `sed` or `perl` (but it's not nearly as useful) and it uses function return values instead of magic variables.

It would be really useful for certain use cases if we could use Deno to create simple Unix-style filters, this is currently the only thing that I still use Perl for on a daily basis.

**Comment by ry** [comment]
And if you had to choose one, what would it be? Deno gets a budget of, say, 10 command line flags. Certainly `-l` and `-n` are out of the question.  Reimplementing `xargs` is out of scope.

**Comment by rsp** [comment]
If I had to choose one, it would be `-e` with no implicit print.

This would be consistent with Node, Perl, Ruby, etc. Even awk doesn't print by default.

**Comment by hayd** [comment]
eval should have the same scope as the repl, so perhaps makes sense to implement -e after that?
(We may also won't to compile ts too.)

**Comment by rsp** [comment]
Plus it's easier to add console.log() when needed than to not print something that is printed by default.

Example:

```
$ node -e '[1,2].forEach(x => console.log(x))'
1
2
```

vs.

```
$ node -p '[1,2].forEach(x => console.log(x))'
1
2
undefined
```

**Comment by bartlomieju** [comment]
@ry are you still interested in this implementing this regarding number of flags we already have?

**Comment by ry** [comment]
Yea... we should probably add it. But considering we will probably add commands like "deno fmt" and "deno info" we might want to do "deno eval '[1,2].forEach(x => console.log(x))'" instead of trying to use `-e` and `-p`... We need to have a separate discussion regarding commands vs flags first.

**Comment by bartlomieju** [comment]
Done in #2102 

CC @ry
